### PR TITLE
Refine Handling of Study Update Impacts

### PR DESCRIPTION
**Pull request by achour94**

### CODE REVIEW

File: src/components/network/use-spreadsheet-equipments.js [hunk 1]
```diff
@@ -5,25 +5,115 @@
...
+            deletedEquipments.forEach(({ equipmentType, equipmentId }) => {
+                if (equipmentId && equipmentType) {
+                    console.info(
+                        'removing equipment with id=',
+                        equipmentId,
+                        ' and type=',
+                        equipmentType,
+                        ' from the network'
+                    );
+                    dispatch(deleteEquipment(equipmentType, equipmentId));
```
Comment (sBouzols) [inline src/components/network/use-spreadsheet-equipments.js:99]: Maybe Add a deleteEquipments in redux action to mange all `deletedEquipments` into one dispatch

File: src/hooks/use-update-study-impacts.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+interface StudyImpactsWithReset extends NetworkImpactsInfos {
+    resetImpactedSubstationsIds: () => void;
+    resetDeletedEquipments: () => void;
+    resetImpactedElementTypes: () => void;
+}
+
+/**
+ * Custom hook that consume the update notification 'study' and return the impacts of the study
+ */
+export const useUpdateStudyImpacts = (): StudyImpactsWithReset => {
```
Comment (sBouzols) [inline src/hooks/use-update-study-impacts.ts:26]: Rename `useGetStudyImpact` maybe ?

File: src/components/network/network-map-tab.jsx [hunk 3]
```diff
@@ -707,22 +709,35 @@ export const NetworkMapTab = ({
...
+                return Promise.reject();
+            }
+
+            const mapEquipmentsTypes = [
+                EQUIPMENT_TYPES.SUBSTATION,
+                EQUIPMENT_TYPES.LINE,
+                EQUIPMENT_TYPES.TIE_LINE,
+                EQUIPMENT_TYPES.HVDC_LINE,
+            ];
+            const impactMapEquipmentTypes = impactedElementTypes?.filter(
```
Comment (sBouzols) [inline src/components/network/network-map-tab.jsx:722]:
```suggestion
            const impactedMapEquipmentTypes = impactedElementTypes?.filter(
```

**Review by sBouzols — APPROVED** [review]
Code Review OK
Tests OK
* multiple deletion
* Load variation
* Generator Variation

Console Warning check OK

File: src/components/network/use-spreadsheet-equipments.js [hunk 4]
```diff
@@ -5,25 +5,115 @@
...
     const studyUuid = useSelector((state) => state.studyUuid);
     const currentNode = useSelector((state) => state.currentTreeNode);
     const [errorMessage, setErrorMessage] = useState();
     const [isFetching, setIsFetching] = useState(false);
 
+    const {
+        impactedSubstationsIds,
+        deletedEquipments,
+        impactedElementTypes,
+        resetImpactedSubstationsIds,
```
[the remark this answers is not in the document]
Comment (achour94) [reply src/components/network/use-spreadsheet-equipments.js:38]: we have a dependency on allEquipments in the useEffect so if we don't reset these values after processing them we risk to do that process multiple times and having errors

**Pull request merged**